src/hevm/src/EVM.hs

```diff
@@ -2693,6 +2751,20 @@ costOfCreate (FeeSchedule {..}) availableGas' hashSize =
     hashCost   = g_sha3word * ceilDiv (num hashSize) 32
     initGas    = allButOne64th (availableGas - createCost)
 
+concreteModexpGasFee :: ByteString -> Integer
+concreteModexpGasFee input = num $ max 200 ((multiplicationComplexity * iterCount) `div` 3)
+  where (lenb, lene, lenm) = parseModexpLength input
+        ez = isZero (96 + lenb) lene input
+        e' = w256 $ word $ LS.toStrict $
+          lazySlice (96 + lenb) (min 32 lene) input
+        nwords = ceiling $ num (max lenb lenm) / 8
+        multiplicationComplexity = nwords * nwords
+        iterCount' | lene <= 32 && ez = 0
+                   | lene <= 32 = num (log2 e')
+                   | e' == 0 = 8 * (lene - 32)
+                   | otherwise = num (log2 e') + 8 * (lene - 32)
```

Comment:
I don't understand these two cases.

In the [EIP](https://eips.ethereum.org/EIPS/eip-2565#specification) I see the following for the final case, which doesn't seem to map onto the code here?

```python
(8 * (exponent_length - 32)) + ((exponent & (2**256 - 1)).bit_length() - 1)
```

Reply:
exponent_length is `lene`, `x.bit_length - 1` is given by `log2 x`, the truncation to a 256 word happens before in `min 32 lene` on L2702